### PR TITLE
RST-4262 Writing out connection header for latched topics

### PR DESCRIPTION
**Pull request by ayrton04**
Addresses https://locusrobotics.atlassian.net/browse/RST-4262

### CODE REVIEW

File: tools/rosbag/src/recorder.cpp [hunk 1]
```diff
@@ -609,7 +609,18 @@ void Recorder::doRecordSnapshotter() {
             OutgoingMessage out = out_queue.queue->front();
             out_queue.queue->pop();
 
-            bag_.write(out.topic, out.time, *out.msg);
+            ros::Time write_stamp = out.time;
```
Comment (ayrton04) [inline tools/rosbag/src/recorder.cpp]: Two things:

- The snapshotter was also missing the connection header getting written out
- The snapshotter was always writing the timestamp of the original message, even if that message was latched and published thousands of seconds ago (like with global costmaps). This made the snapshot bags hard to use for nav.

Comment (tappan-at-git) [reply tools/rosbag/src/recorder.cpp]: A part of me is worried about nodes tripping over updated timestamps (e.g. if we're replaying multiple sequential bags), but I don't know any specific nodes that would have issues.

File: tools/rosbag/src/recorder.cpp [hunk 2]
```diff
@@ -605,11 +605,25 @@ void Recorder::doRecordSnapshotter() {
...
+        ros::Time now = ros::Time::now();
+
         while (!out_queue.queue->empty()) {
             OutgoingMessage out = out_queue.queue->front();
             out_queue.queue->pop();
 
-            bag_.write(out.topic, out.time, *out.msg);
+            ros::Time write_stamp = out.time;
+
+            // If a snaptshot topic is also latched, handle its timestamp
```
Comment (tappan-at-git) [inline tools/rosbag/src/recorder.cpp]:
```suggestion
            // If a snapshot topic is also latched, handle its timestamp
```

**Review by tappan-at-git — APPROVED** [review]

File: tools/rosbag/src/recorder.cpp (hunk 1 above)
Comment (ayrton04) [reply tools/rosbag/src/recorder.cpp]: How does that differ from this?

https://github.com/locusrobotics/ros_comm/blob/eae18e0e37871518bebcebecc68ea1aa329fe82b/tools/rosbag/src/recorder.cpp#L413-L418

Comment (tappan-at-git) [reply tools/rosbag/src/recorder.cpp]: Oh, huh. That block is rewriting rosbag's record of when the message was received. I misread this section as applying to the actual message headers, but obviously that's not what's actually happening on second look.

**Review by abencz — APPROVED** [review]

**Comment by ayrton04** [comment]
Turns out snapshotting isn't actually supported in this package. That was deprecated, which is why we use `rosbag_snapshot`.

**Pull request merged**